lib/matplotlib/widgets.py

```diff
@@ -2048,16 +2048,36 @@ def _on_key_release(self, event):
         """Key release event handler."""
 
     def set_visible(self, visible):
-        """Set the visibility of our artists."""
-        self.visible = visible
+        """Set the visibility of the selector artists."""
+        self._visible = visible
         for artist in self.artists:
             artist.set_visible(visible)
 
-    def clear(self):
-        """Clear the selection and set the selector ready to make a new one."""
+    def get_visible(self):
+        """Get the visibility of the selector artists."""
+        return self._visible
+
+    visible = _api.deprecated("3.6", name="visible",
```

Comment:
The visible getter should not be deprecated, only the setter. This is the least API change and we have not yet decided whether we may add more properties in the future.
We must however deprecate the setter because it has not been working and we cannot guess whether the user wants the existing broken behavior or the semantically correct one.